code/modules/events/antagonist_critter_ghost_respawn.dm

```diff
@@ -55,17 +76,25 @@
 
 			var/atom/pestlandmark = pick(EV)
 
-			var/list/select = pick(pest_invasion_critter_types)
-			if (source)
-				if (source == "fire_elemental")
+			var/list/select = null
+			switch (source)
+				if (null) //random
+					select = pick(src.pest_invasion_critter_types)
+				if ("fire_elemental")
```

Comment:
same down here, remove the fire_elemental spider and gunbot checks

Reply:
I have made the requested changes